### PR TITLE
wheel.ino: debug logging, code cleanup, Pot class, arrays, and more

### PR DESCRIPTION
**Pull request by danapolsUM**
## Improvements to ```wheel.ino``` and ```Mapped_Encoder```

- Debugging modes for outputting
- Arrays to iterate through buttons and LEDs with for loops
- Simplified logic
- Removed magic variables
- Removed unused variables
- Code now only waits for ```Serial``` to start in debugging mode
- Error avoidance for faulty received CAN messages
- Mapped_Encoder Library with its own class to make calibration and general usage of encoder and throttle easier
- Fixed floatMap, tested with math but not yet in code

## Future plans

- Calibration logic for potentiometers, buttons, and LEDs
- Saving calibration settings to [EEPROM](https://docs.arduino.cc/learn/built-in-libraries/eeprom/)
- Sending logging data to shore
- Onshore testing
- CAN testing
- Water testing
- Calibration during crisis testing
- Hardware issues automatic detection and warning system
- Calibration integration with display
- Unit testing

### CODE REVIEW

File: wheel/wheel.ino [hunk 1]
```diff
@@ -39,208 +84,213 @@ GND	| GND
...
+
+// CAN Ids
+#define encoderId  1999
+#define throttleId 1998
+#define buttonsId  2000
+#define LEDsId     2001
+
+
+// For logging
+#include <Arduino_DebugUtils.h>
```
Comment (msharash) [inline wheel/wheel.ino]: top of file

File: wheel/wheel.ino [hunk 2]
```diff
@@ -39,208 +84,213 @@ GND	| GND
...
+  } else {
+    DEBUG_INFO("All data sent");
+  }
+  DEBUG_VERBOSE("endPacket output: %d", end);
 
 }
 
-float readThrottle(){
-  return floatMap(analogRead(ThrottlePin), 0, 4095, 0, 100); //Scale => [0,100]
+void encoderSend() {
```
Comment (msharash) [inline wheel/wheel.ino]: why have a function that just calls another func

File: wheel/wheel.ino [hunk 3]
```diff
@@ -39,208 +84,213 @@ GND	| GND
...
 
-void encodertoCAN(int val){
-  CAN.beginPacket(1999);
-  char data[sizeof(val)];               //Create char array
-  memcpy(data, &val, sizeof(val));       //Store bytes of val to array
-  for(int j = 0; j < sizeof(val); j++){  //Write bytes one by one to CAN
-    CAN.write(data[j]);
-  }
-  CAN.endPacket();
+void throttleSend() {
```
Comment (msharash) [inline wheel/wheel.ino]: same thing

File: wheel/wheel.ino [hunk 4]
```diff
@@ -31,6 +65,17 @@ E-stop (A-BRB) is intentionally not included
 #define LED5 18
 #define LED6 17
 
+// Lists that contain the Steering Wheel buttons and the LED pins
+#define NUM_BUTTONS 5
+#define NUM_LEDS 6
+
+int buttons[NUM_BUTTONS] = {CommsPin, TrimUpPin, TrimDownPin, GainUpPin, GainDownPin};
```
Comment (msharash) [inline wheel/wheel.ino:64]: define ints and const ints after #defines, just an organizational thing

File: wheel/wheel.ino [hunk 5]
```diff
@@ -31,6 +65,17 @@ E-stop (A-BRB) is intentionally not included
...
+// Lists that contain the Steering Wheel buttons and the LED pins
+#define NUM_BUTTONS 5
+#define NUM_LEDS 6
+
+int buttons[NUM_BUTTONS] = {CommsPin, TrimUpPin, TrimDownPin, GainUpPin, GainDownPin};
+const String button_names[NUM_BUTTONS] = {"Comms", "Trim Up", "Trim Down", "Gain Up", "Gain Down"};
+
+int LEDs[NUM_LEDS] = {LED1, LED2, LED3, LED4, LED5, LED6};
+
+#include <CAN.h>
```
Comment (msharash) [inline wheel/wheel.ino]: top of file

**Review by msharash — CHANGES_REQUESTED** [review]
Look at things I said, dailey is addressing floatmap things so have fun there. Also still think we can do this w like an analog read and like one line of hard coded math but up2u

File: wheel/wheel.ino (hunk 2 above)
Comment (danapolsUM) [reply wheel/wheel.ino]: So that I can pass different values depending on the potentiometer in the future, but don't need to.

**Pull request closed without merging**